pages/checkout/dropin.vue

```diff
@@ -0,0 +1,151 @@
+<template>
+  <div>
+    <div id="payment-page">
+      <div class="container">
+        <div class="payment-container">
+          <div id="dropin-container" class="payment"></div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup>
+
+import { ref, onMounted } from 'vue';
+import { AdyenCheckout, Dropin } from '@adyen/adyen-web/auto';
+
+// Reactive references
+const sessionId = ref('');
+const redirectResult = ref('');
+
+// Function to invoke server-side API
+async function sendPostRequest(url, data) {
+  const res = await fetch(url, {
+    method: "POST",
+    body: data ? JSON.stringify(data) : "",
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+  return await res.json();
+}
+
+// Function to handle the server response
+async function handleServerResponse(res, component) {
+  if (res.action) {
+    component.handleAction(res.action);
+  } else {
+    switch (res.resultCode) {
+      case "Authorised":
+        window.location.href = "/result/success";
+        break;
+      case "Pending":
+      case "Received":
+        window.location.href = "/result/pending";
+        break;
+      case "Refused":
+        window.location.href = "/result/failed";
+        break;
+      default:
+        window.location.href = "/result/error";
+        break;
+    }
+  }
+}
+
+// Function to create AdyenCheckout instance
+async function createAdyenCheckout(session, clientKey) {
+  return AdyenCheckout(
+    {
+      clientKey,
+      session: session,
+      environment: "test",
+      amount: {
+        value: 10000,
+        currency: 'EUR'
+      },
+      locale: "en-US",
+      countryCode: 'NL',
+      showPayButton: true,
+      // override Security Code label
+      translations: {
+        'en-US': {
+            'creditCard.securityCode.label': 'CVV/CVC'
+        }
+      },
+      onPaymentCompleted: (result, component) => {
+        console.info("onPaymentCompleted");
+        console.info(result, component);
+        handleServerResponse(result, component);
+      },
+      onPaymentFailed: (result, component) => {
+        console.info("onPaymentFailed");
+        console.info(result, component);
+        handleServerResponse(result, component);
+      },
+      onError: (error, component) => {
+        console.error("onError");
+        console.error(error.name, error.message, error.stack, component);
+        handleServerResponse(error, component);
+      },
+    }
+  );
+}
+
+// Function to start checkout
+async function startCheckout() {
+  try {
+    const { response, clientKey } = await sendPostRequest(`/api/sessions`);
```

Comment:
Question: It's kind of confusing, I see that `/sessions` returns the `clientKey` I see that in the server-call, we're returning this value. Is this considered the best practice? I'd separate these concerns